Certification/Loader.php

```diff
@@ -43,16 +43,15 @@ public static function init($number, array $categories, $path)
             return new Set(array());
         }
 
-        $dataMax = count($data) - 1;
+        $dataMax = count($data);
+
+        // Number must not be superior than the number of questions
+        $number = min($number, $dataMax);
 
         $questions = array();
 
         for ($i = 0; $i < $number; $i++) {
-            do {
-                $random = rand(0, $dataMax);
-            } while (isset($questions[$random]) && count($questions) < $dataMax);
-
-            $item = $data[$random];
+            $item = array_splice($data, array_rand($data), 1)[0];
```

Comment:
it's smart, I'm interested on which bug you tried to fix with this patch, this is why I asked for a test to avoid any regression 👍